climpred/prediction.py

```diff
@@ -135,168 +124,68 @@ def compute_reference(ds,
-    else:
-        return skill
-
-
-def compute_persistence_pm(ds, control, nlags=None, metric='pearson_r',
-                           dim='time', init_month_index=0):
-    """
-    Computes the skill of  a persistence forecast from a control run.
-
-    This simply applies some metric on the input out to some lag. The user
-    should avoid computing persistence with prebuilt ACF functions in e.g.,
-    python, MATLAB, R as they tend to use FFT methods for speed but incorporate
-    error due to this.
-
-    TODO: Merge this and `compute_persistence` into one function. These two
-    functions employ different philosophies on how to compute persistence.
-
-    Currently supported metrics for persistence:
-    * pearson_r
-    * rmse
-    * mse
-    * mae
-
-    Reference:
-    * Chapter 8 (Short-Term Climate Prediction) in
-        Van den Dool, Huug. Empirical methods in short-term climate prediction.
-        Oxford University Press, 2007.
 
-    Args:
-        ds (xarray object): The initialization years to get persistence from.
-        reference (xarray object): The reference time series.
-        nlags (int, default None): Number of lags to compute persistence to.
-            If None, use the number of lead steps from the prediction
-            ensemble.
-        metric (str): Metric name to apply at each lag for the persistence
-                      computation. Default: 'pearson_r'
-        dim (str): Dimension over which to compute persistence forecast.
-                   Default: 'time'
-
-    Returns:
-        pers (xarray object): Results of persistence forecast with the input
-                              metric applied.
-    """
-    check_xarray(control)
     metric = get_metric_function(metric)
-    if nlags is None:
-        nlags = ds.lead.size
-
-    if metric not in [_pearson_r, _rmse, _mse, _mae]:
-        raise ValueError("""Please select between the following metrics:
-            'pearson_r',
-            'rmse',
-            'mse',
-            'mae'""")
-
-    init_years = ds['init'].values
-    init_years = _intersection(init_years, list(control.time.values))
-    if isinstance(ds.init.values[0],
-                  cftime._cftime.DatetimeProlepticGregorian) \
-            or isinstance(ds.init.values[0], np.datetime64):
-        init_cftimes = []
-        for year in init_years:
-            init_cftimes.append(control.sel(
-                time=str(year)).isel(time=init_month_index).time)
-        init_cftimes = xr.concat(init_cftimes, 'time')
-    elif isinstance(ds.init.values[0], np.int64):
-        init_cftimes = []
-        for year in init_years:
-            init_cftimes.append(control.sel(
-                time=year).time)
-        init_cftimes = xr.concat(init_cftimes, 'time')
-    else:
-        raise ValueError(
-            'Set time axis to xr.cftime_range, pd.date_range or np.int64.')
+    if metric not in all_metrics:
+        raise ValueError("""Please input metric from all_metrics.""")
```

Comment:
all_metrics can be parametrized here

Also, since it's a single line, maybe just use single apostrophes/quotes?
`raise ValueError(f'Please input a metric from the following {ALL_METRIC_STRINGS}')`